### DISCUSSION TITLE
Adding support for React.lazy() and Suspense

### DISCUSSION THREAD

**Discussion by 4x4notfound** [remix-run/remix#5285] [General]
In particular environments like Cloudflare workers the `@remix-run/dev/server-build` creates a long startup time (sus is react router) which causes a failure for deployment.

I think adding [code splitting](https://reactjs.org/docs/code-splitting.html) techniques like `React.lazy` with Suspense would be helpful in supporting these restrictive envs.

**Comment by sergiodxa** [comment]
If you’re using React 18 you can do that, but I AFAIK Cloudflare runtimes doesn’t support dynamic imports so here the limit is not in Remix or React but rather Cloudflare.

**Reply by 4x4notfound** [reply]
Thanks @sergiodxa. I found your blog post on lazy loading with react 18 and remix and it inspired an approach that works but is not ideal due to a flash of unstyled content and some hacks to get the lazy loaded hooks/effects to work . All because one dependency results in a large startup time with non-deterministic deployment errors 🤷 

Also working with the team at CF to extend startup time from 200ms to 300ms or to resolve this problem.
